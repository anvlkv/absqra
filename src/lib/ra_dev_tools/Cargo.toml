[package]
name = "ra_dev_tools"
version = "0.1.0"
authors = ["a.nvlkv <a.nvlkv@protonmail.ch>"]
edition = "2018"
build = "build.rs"

[build-dependencies]
build-deps = "0.1.4"
# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
insta = "0.16.0"
regex = "1.3.9"
example_files_macro = {path="./src/lib/example_files_macro"}
files_iterator = {path="./src/lib/files_iterator"}
