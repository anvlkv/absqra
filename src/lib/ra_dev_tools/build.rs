extern crate build_deps;

fn main() {
    // Enumerate files in sub-folder "examples/*", being relevant for the test-generation (as example)
    // If function returns with error, exit with error message.
    build_deps::rerun_if_changed_paths( "../../../examples/*/*" ).unwrap();

    // Adding the parent directory "examples" to the watch-list will capture new-files being added
    build_deps::rerun_if_changed_paths( "../../../examples/*" ).unwrap();
}