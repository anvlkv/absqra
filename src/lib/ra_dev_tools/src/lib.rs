pub extern crate insta;
pub use files_iterator::for_each_ra_example_file;

#[allow(unused_attributes)]
#[macro_use] pub use example_files_macro::make_example_tests;

#[cfg(test)]
mod test {
    use super::make_example_tests;

    #[make_example_tests]
    #[test]
    fn it_works_with(content: String, file_name: String) {
        println!("didn't panic with content: \n[{}] of file_name: {}", content, file_name);
    }
}