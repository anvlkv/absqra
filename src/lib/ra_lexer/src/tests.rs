use super::*;
use ra_dev_tools::insta::assert_json_snapshot;
use ra_dev_tools::make_example_tests;

#[make_example_tests]
#[test]
fn it_should_match_snapshots(contents: String, file_name: String) {
    let tokens: Vec<RaToken> = tokenize(&contents).map(|r| r.unwrap()).collect();
    assert_json_snapshot!(file_name, tokens)
}