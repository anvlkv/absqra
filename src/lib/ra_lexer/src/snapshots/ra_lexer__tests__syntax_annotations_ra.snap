---
source: src/lib/ra_lexer/src/tests.rs
expression: tokens
---
[
  {
    "kind": "HashPound",
    "position": [
      [
        1,
        1
      ],
      [
        1,
        2
      ]
    ],
    "level": 1
  },
  {
    "kind": {
      "Identifier": "annotation"
    },
    "position": [
      [
        1,
        2
      ],
      [
        1,
        12
      ]
    ],
    "level": 1
  },
  {
    "kind": {
      "Comment": [
        " this is annotation",
        false
      ]
    },
    "position": [
      [
        1,
        13
      ],
      [
        2,
        0
      ]
    ],
    "level": 1
  },
  {
    "kind": "HashPound",
    "position": [
      [
        3,
        1
      ],
      [
        3,
        2
      ]
    ],
    "level": 1
  },
  {
    "kind": {
      "Identifier": "media_context"
    },
    "position": [
      [
        3,
        2
      ],
      [
        3,
        15
      ]
    ],
    "level": 1
  },
  {
    "kind": "Colon",
    "position": [
      [
        3,
        15
      ],
      [
        3,
        16
      ]
    ],
    "level": 1
  },
  {
    "kind": {
      "Identifier": "media_option"
    },
    "position": [
      [
        3,
        16
      ],
      [
        3,
        28
      ]
    ],
    "level": 1
  },
  {
    "kind": {
      "Comment": [
        " ...[:media_option] more options possible",
        false
      ]
    },
    "position": [
      [
        3,
        29
      ],
      [
        4,
        0
      ]
    ],
    "level": 1
  },
  {
    "kind": "HashPound",
    "position": [
      [
        5,
        1
      ],
      [
        5,
        2
      ]
    ],
    "level": 1
  },
  {
    "kind": {
      "Identifier": "annotation"
    },
    "position": [
      [
        5,
        2
      ],
      [
        5,
        12
      ]
    ],
    "level": 1
  },
  {
    "kind": "Colon",
    "position": [
      [
        5,
        12
      ],
      [
        5,
        13
      ]
    ],
    "level": 1
  },
  {
    "kind": {
      "Identifier": "option_a"
    },
    "position": [
      [
        5,
        13
      ],
      [
        5,
        21
      ]
    ],
    "level": 1
  },
  {
    "kind": "Colon",
    "position": [
      [
        5,
        21
      ],
      [
        5,
        22
      ]
    ],
    "level": 1
  },
  {
    "kind": {
      "Identifier": "option_b"
    },
    "position": [
      [
        5,
        22
      ],
      [
        5,
        30
      ]
    ],
    "level": 1
  },
  {
    "kind": "Colon",
    "position": [
      [
        5,
        30
      ],
      [
        5,
        31
      ]
    ],
    "level": 1
  },
  {
    "kind": {
      "Identifier": "option_c"
    },
    "position": [
      [
        5,
        31
      ],
      [
        5,
        39
      ]
    ],
    "level": 1
  },
  {
    "kind": "Colon",
    "position": [
      [
        5,
        39
      ],
      [
        5,
        40
      ]
    ],
    "level": 1
  },
  {
    "kind": {
      "Identifier": "option_d"
    },
    "position": [
      [
        5,
        40
      ],
      [
        5,
        47
      ]
    ],
    "level": 1
  }
]
